python/afdko/otfautohint/hinter.py

```diff
@@ -459,29 +493,30 @@ def addSegment(self, fr, to, loc, pe1, pe2, typ, desc, mid=False):
 
         if not pe1 and not pe2:
             return
+        assert self.hs is not None
         self.hs.addSegment(fr, to, loc, pe1, pe2, typ, self.Bonus,
                            self.isV(), mid1, mid2, desc)
 
-    def CPFrom(self, cp2, cp3):
+    def CPFrom(self, cp2, cp3) -> Any:
         """Return point cp3 adjusted relative to cp2 by CPFrac"""
         return (cp3 - cp2) * (1.0 - self.CPfrac) + cp2
 
-    def CPTo(self, cp0, cp1):
+    def CPTo(self, cp0, cp1) -> Any:
```

Comment:
What is the purpose of the Any return type notations? Does it help compared with just leaving them off?

Reply:
Python typing is progressive, and basically, I haven't finished adding all the annotations. But if you add *any* type notation, a method gets checked as best it can be, but with no type notation at all you don't get any checking or inferencing.